api/scpca_portal/s3.py

```diff
@@ -242,3 +214,21 @@ def generate_pre_signed_link(filename: str, key: str, bucket_name: str) -> str:
         },
         ExpiresIn=60 * 60 * 24 * 7,  # 7 days in seconds.
     )
+
+
+class DownloadableFiles:
+    def __init__(self, original_files: List[OriginalFile]):
+        self._original_files = original_files
+
+    def _is_downloadable(self, original_file):
+        return not Path(original_file.s3_key).exists()
+
+    @property
+    def bucket_paths(self):
+        bucket_paths = defaultdict(list)
+        for original_file in self._original_files:
+            if self._is_downloadable(original_file):
+                bucket_path = Path(original_file.s3_bucket) / original_file.download_dir
+                bucket_paths[bucket_path].append(original_file)
```

Comment:
```suggestion
                bucket_paths[(bucket_path, download_dir)].append(original_file.download_path)
```

Can we keep the logic for determining the download dir in here so that it is co-located?